modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/subscription/SubscriptionManagerImpl.kt

```diff
@@ -281,11 +281,15 @@ class SubscriptionManagerImpl @Inject constructor(
             LogBuffer.e(LogBuffer.TAG_SUBSCRIPTIONS, "expected 1 product when sending purchase to server, but there were ${purchase.products.size}")
         }
 
-        val response = syncManager.subscriptionPurchase(SubscriptionPurchaseRequest(purchase.purchaseToken, purchase.products.first())).await()
-        val newStatus = response.toStatus()
-        cachedSubscriptionStatus = newStatus
-        subscriptionStatus.accept(Optional.of(newStatus))
-        purchaseEvents.accept(PurchaseEvent.Success)
+        try {
+            val response = syncManager.subscriptionPurchase(SubscriptionPurchaseRequest(purchase.purchaseToken, purchase.products.first())).await()
+            val newStatus = response.toStatus()
+            cachedSubscriptionStatus = newStatus
+            subscriptionStatus.accept(Optional.of(newStatus))
+            purchaseEvents.accept(PurchaseEvent.Success)
+        } catch (ex: Exception) {
+            LogBuffer.e(LogBuffer.TAG_BACKGROUND_TASKS, ex, "Failed to send purchase to server.")
+        }
```

Comment:
Good point. I was trying to figure out why the app was crashing with the error 'UndeliverableException: The exception could not be delivered to the consumer because it has already canceled/disposed...' and left that in there. I will remove the try catch.

Reply:
> I was trying to figure out why the app was crashing with the error 'UndeliverableException: The exception could not be delivered to the consumer because it has already canceled/disposed...'

The crashing part of that was because I accidentally messed up how we were catching this exception before on teh phone app. I've put up [a PR](https://github.com/Automattic/pocket-casts-android/pull/1104) with a fix and some more context on that (does not address the cause of the exception though).